h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -323,7 +317,13 @@ private static DMatrix dense(Chunk[] chunks, int weight, DataInfo di, int respId
         float[][] data = allocateDenseMatrix(chunks[0].len(), di);
 
         long actualRows = denseChunk(data, chunks, weight, respIdx, di, cols, resp, weights);
-        assert actualRows == chunks[0].len();
+        if (weight == -1) {
+            assert actualRows == chunks[0].len();
+        } else {
+            // Rows with weights zero are skipped, thus the resulting number of rows must be equal
+            // to number of non-zero weights in weights chunk
+            assert actualRows == weights.length;
```

Comment:
What about ? `assert actualRows == di.getWeightsVec().nzCnt()` here ? This `di.getWeightsVec().nzCnt()` method returns different number then `weights.length` in some cases ,which is really strange and the assert fails in such case. The `weights` array is allocated after manually counting number of non-zero elements in the weight vector - this way the assert always works. 

I wonder if I'm missing something.